raytracing/rays.py

```diff
@@ -284,18 +286,14 @@ def __init__(self, yMax=1.0, yMin=None, thetaMax=pi / 2, thetaMin=None, maxCount
         self.thetaMin = thetaMin
         if thetaMin is None:
             self.thetaMin = -thetaMax
-        super(RandomRays, self).__init__(rays=None)
+        super(RandomRays, self).__init__()  # Even with this, rays = []
```

Comment:
Either way, RandomRays need to provide a protected access to __getitem__ without exposing _rays.
If rays have already been produced, take them from array, if they have. not, generate up to the item and return.